tests/plugins/dut_monitor/dut_monitor.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import time
+import os
+
+from  datetime import datetime
+
+
+FETCH_CPU_CMD = "ps -eo pcpu,args | sort -k 1 -r | tail -n +2"
+FETCH_HDD_CMD = "df -hm /"
+MEASURE_DELAY = 2
+
+
+def process_cpu(log_file):
+    """
+    @summary: Fetch CPU utilization. Write to the file total and top 10 process CPU utilization.
+    @param log_file: Opened file object to store fetched CPU utilization.
+    """
+    buff = []
+    general_template = """
+"{timestamp}":
+    total: {total}
+    top_consumer:
+"""
+    per_process_template = "        {cpu_utilization}: \"{process}\""
+    top_consumer_list = []
+    total = 0
+    top_consumer_counter = 10
+
+    for line in os.popen(FETCH_CPU_CMD).readlines():
+        process_consumed = float(line.split()[0])
+        total += process_consumed
+        if top_consumer_counter:
```

Comment:
Since the command output is already sorted, you can do it in a more pythonic way:
```
for line in os.popen(FETCH_CPU_CMD).readlines()[:top_consumer_counter]:
```
Then there is no need to decrement top_consumer_counter and check its value in the loop.

Reply:
But in such case total CPU utilization will not be taken into account. Here is calculating total CPU utilization and top 10 CPU consumer.

Reply:
I mean change the code like below:
```python
    per_process_template = "        {cpu_utilization}: \"{process}\""
    top_consumer_list = []
    total = 0
    top_consumer_counter = 10

    for line in os.popen(FETCH_CPU_CMD).readlines()[:top_consumer_counter]:
        process_consumed = float(line.split()[0])
        total += process_consumed
        top_consumer_list.append(per_process_template.format(cpu_utilization=process_consumed, process=line.split()[1]))
```
The total calculation is kept. Only to get rid of the code for decrementing top_consumer_counter.

Reply:
Maybe you confused about what this loop should collect.
The idea of this loop is to collect CPU utilization of ALL processes which are running on the DUT, NOT ONLY top 10 consumers. If loop will use ".readlines()[:top_consumer_counter]", it means that it will store CPU utilization only from first 10 processes.

So "total" variable will not have correct information about total CPU utilization on the DUT.
I suggest to leave this as it is.

Your thoughts?

Reply:
Ah, I see, you are right. I was confused. Sorry! Now it's clear.

Reply:
Ok, no problem.